packages/serverless-components/nextjs-cdk-construct/src/index.ts

```diff
@@ -116,11 +130,16 @@ export class NextJSLambdaEdge extends cdk.Construct {
         "RegenerationFunction",
         {
           handler: "index.handler",
-          runtime: lambda.Runtime.NODEJS_14_X,
-          timeout: Duration.seconds(30),
           code: lambda.Code.fromAsset(
             path.join(this.props.serverlessBuildOutDir, "regeneration-lambda")
-          )
+          ),
+          role: this.edgeLambdaRole,
+          runtime:
+            toLambdaOption("regenerationLambda", props.runtime) ??
+            lambda.Runtime.NODEJS_12_X,
```

Comment:
This should remain NODEJS_14_X?

Reply:
I'm not 100% sure why it was hardcoded as NODEJS_14_X but I think it's unintuitive that the regeneration function is the only one with a fixed nodejs version. This might result in some weird bugs if it is the regeneration function uses a different runtime than the default function because they use the same js page bundles. This line I added is the same as for the other functions qnd it makes sure that all functions use the same nodejs version except if the user really wants to set something different throught props

Reply:
Don't we want the regeneration lamda have the same role as the other functions? I know in my case I need this because the role grants access to dynamo to get the date necessary for generating a page

Reply:
Sorry @mathisobadia to be clear I think what @dmsolutionz is saying is that the "default should remain as NODEJS_14_X" rather than it shouldn't be editable. If you change to `NODEJS_12_X` as the default that's a breaking change.
Having it as editable like the others is exactly the desired outcome here.

Reply:
Oh yeah that makes perfect sense! I've updated my pull request accordingly.